src/rulesets/oas/functions/utils.ts/unusedDefinitionOrComponentsSchema.ts

```diff
@@ -0,0 +1,35 @@
+import { DepGraph } from 'dependency-graph';
+import { IFunctionResult } from '../../../../types';
+
+export type OasReusableObjectsLocation = '#/definitions' | '#/components/schemas';
+
+export const unusedDefinitionOrComponentsSchema = (
```

Comment:
As you rightfully noticed, the function could accept any path.
Whatcha think about moving it outside of oas functions? This could be placed in a core functions directory, so that each rule can use it, regardless of spec.

Reply:
@P0lip Ok. What would you like it to return? An array of strings (eg. `["#/some/where", ...]`)? An array of path segments (eg. `[ [ 'some', 'where'], ...]`)?

I'll try and come up with a name for it (but that's not my sweet spot. I'm slightly better at defining cache invalidation policies, though :wink: )

Reply:
The function must follow `IFunction`, so we have to return `IFunctionResult` or nothing>
Suggest we return a path to the issue (`['some', 'where']`) and some message, which could be potentially customizable using functionOptions. What do you think?

> I'll try and come up with a name for it (but that's not my sweet spot. I'm slightly better at defining cache invalidation policies, though wink )

Hm, what about `unreferencedProperty` or something more or less like this :laughing: 
I'm not perfect at naming either, so take my proposal with a grain of salt.

Reply:
Oh! I initially misunderstood you. I thought that you were willing to get an oas and a core utility function (which explains my question above with regards to the expected output type of the utility function). 

I'll fix my proposal accordingly to your last statement.

Reply:
>I'll fix my proposal accordingly to your last statement.

Fixed